BuildScript.Nuke/BaseBuild.Documentation.cs

```diff
@@ -0,0 +1,220 @@
+                                      { SandcastleProperties.VendorName, AssemblyMetadata.CompanyName },
+                                      {
+                                          SandcastleProperties.BuildLogFile,
+                                          $"{Directories.Log}Sandcastle.{AssemblyMetadata.DocumentationFileName}.log"
+                                      },
+                                      { SandcastleProperties.WorkingPath, Directories.DocumentationBaseDirectory / "Working" },
+                                      { SandcastleProperties.SHFBROOT, sandcastlePackage.Directory / "tools" },
+                                      { SandcastleProperties.ComponentPath, $"{sandcastlePackage.Directory.Parent.Parent}" }
+                                  };
+    documentationProperties.ForEach(
+        keyValue =>
+        {
+          XmlTasks.XmlPoke(
+              documentationSandcastleProjectFile,
+              $"/mb:Project/mb:PropertyGroup/mb:{keyValue.Key}",
+              keyValue.Value,
+              ("mb", "http://schemas.microsoft.com/developer/msbuild/2003")
+          );
+        });
+    var sandcastleProjectBuilder = new SandcastleProjectBuilder(
+        documentationSandcastleProjectFile,
+        documentationProjectFiles,
+        namespaceSummaryFiles);
+    sandcastleProjectBuilder.UpdateSandcastleProject();
+    MSBuildTasks.MSBuild(settings => settings
+        .SetTargetPath(documentationSandcastleProjectFile)
+        .SetProcessEnvironmentVariable(SandcastleProperties.SHFBROOT, sandcastlePackage.Directory / "tools")
+    );
+    DocumentOutputFile = @$"{documentationCompilationOutputDirectory}\{AssemblyMetadata.DocumentationFileName}.chm";
+  }
+
+  private void CreateDocumentationNugetPackage ()
+  {
+    var documentationProjects = ReleaseProjectFiles.Where(projectMetadata => projectMetadata.IsDocumentationFile).ToList();
+    if (documentationProjects.Count == 0)
+      Log.Warning("No documentation projects found!");
+    documentationProjects.ForEach(projectFile =>
+    {
+      PrepareDocumentationProjectFile(projectFile);
+      try
+      {
+        GenerateSinglePackageWithDebugSymbols(projectFile);
+      }
+      catch (Exception ex)
+      {
+        Log.Error(ex, "Error while generating nuget package for documentation");
+      }
+      finally
+      {
+        RestoreDocumentationProjectFile(projectFile);
+      }
+    });
+  }
+
+  private void RestoreDocumentationProjectFile (ProjectMetadata projectFile)
+  {
+    FileSystemTasks.CopyFile($"{projectFile.ProjectPath}.bakup", projectFile.ProjectPath, FileExistsPolicy.Overwrite);
+    FileSystemTasks.DeleteFile($"{projectFile.ProjectPath}.bakup");
+  }
+
+  private void PrepareDocumentationProjectFile (ProjectMetadata projectFile)
+  {
+    FileSystemTasks.CopyFile(projectFile.ProjectPath, $"{projectFile.ProjectPath}.bakup", FileExistsPolicy.Overwrite);
+    var xmlDocument = new XmlDocument();
```

Comment:
Why are you using XmlPoke above and XmlDocument parsing here? Is there anything that one can handle and the other alternative can't? I would advocate for using one of those in the whole file instead of mixing both if possible.